parsl/executors/high_throughput/executor.py

```diff
@@ -714,8 +714,15 @@ def scale_in(self, blocks=None, block_ids=[], force=True, max_idletime=None):
 
         return self._filter_scale_in_ids(to_kill, r)
 
-    def _get_job_ids(self) -> List[object]:
-        return list(self.blocks.values())
+    def _get_block_and_job_ids(self) -> Tuple[List[object], List[object]]:
+        # Not using self.blocks.keys() and self.blocks.values()
+        # The dictionary may be changed during invoking this function
+        # As scale_in and scale_out are invoked in multiple threads
+        block_ids = list(self.blocks.keys())
+        job_ids = []
+        for bid in block_ids:
+            job_ids.append(self.blocks[bid])
+        return block_ids, job_ids
```

Comment:
There is a mypy error here. Here are two questions:

1) what is the type of block IDs? (I think probably str, not object)

2) what is the type of job IDs? (maybe also always a str?)


This fixes mypy for me, but it might not be the right fix:
diff --git a/parsl/executors/high_throughput/executor.py b/parsl/executors/high_throughput/executor.py
index 54f3d19ba..dc8285c29 100644
--- a/parsl/executors/high_throughput/executor.py
+++ b/parsl/executors/high_throughput/executor.py
@@ -714,12 +714,12 @@ class HighThroughputExecutor(StatusHandlingExecutor, RepresentationMixin):
 
         return self._filter_scale_in_ids(to_kill, r)
 
-    def _get_block_and_job_ids(self) -> Tuple[List[object], List[object]]:
+    def _get_block_and_job_ids(self) -> Tuple[List[str], List[object]]:
         # Not using self.blocks.keys() and self.blocks.values()
         # The dictionary may be changed during invoking this function
         # As scale_in and scale_out are invoked in multiple threads
         block_ids = list(self.blocks.keys())
-        job_ids = []
+        job_ids = []  # type: List[object]
         for bid in block_ids:
             job_ids.append(self.blocks[bid])
         return block_ids, job_ids
diff --git a/parsl/executors/status_handling.py b/parsl/executors/status_handling.py
index 994d30c7d..f27c891e7 100644
--- a/parsl/executors/status_handling.py
+++ b/parsl/executors/status_handling.py
@@ -52,7 +52,7 @@ class StatusHandlingExecutor(ParslExecutor):
             return self._provider.status_polling_interval
 
     @abstractmethod
-    def _get_block_and_job_ids(self) -> Tuple[List[object], List[object]]:
+    def _get_block_and_job_ids(self) -> Tuple[List[str], List[object]]:
         raise NotImplementedError("Classes inheriting from StatusHandlingExecutor must implement "
                                   "_get_block_and_job_ids()")